src/web/app/components/teammates-common/submission-status-tooltip.pipe.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { FeedbackSessionSubmissionStatus } from '../../../types/api-output';
+import { SubmissionStatusTooltipPipe } from './submission-status-tooltip.pipe';
+
+describe('SubmissionStatusTooltipPipe', () => {
+  it('create an instance', () => {
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+    expect(pipe).toBeTruthy();
+  });
+
+  it('transform with the tooltip description correctly', () => {
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+    expect(pipe).toBeTruthy();
+  });
+
+  it('transform with no deadlines correctly', () => {
+    jest.useFakeTimers().setSystemTime(new Date('2020-01-01').getTime());
+    const hasNoDeadlines = {
+      studentDeadlines: {},
+      instructorDeadlines: {},
+    };
+    const hasNoOngoingDeadlines = {
+      studentDeadlines: { nonOngoingExtension1: new Date('2019-01-01').valueOf() },
+      instructorDeadlines: { nonOngoingExtension2: new Date('2019-02-01').valueOf() },
+    };
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+
+    expect(pipe.transform(FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN, hasNoDeadlines)).toEqual(
+      pipe.transform(FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN),
+    );
+
+    expect(pipe.transform(FeedbackSessionSubmissionStatus.OPEN, hasNoOngoingDeadlines)).toEqual(
+      pipe.transform(FeedbackSessionSubmissionStatus.OPEN),
+    );
+  });
+
+  it('transform with deadlines correctly', () => {
+    jest.useFakeTimers().setSystemTime(new Date('2020-01-01').getTime());
+    const hasOngoingDeadlines = {
+      studentDeadlines: { ongoingDeadline: new Date('2021-01-01').valueOf() },
+      instructorDeadlines: { nonOngoingDeadline: new Date('2019-01-01').valueOf() },
+    };
+
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+    const extensionMessage = 'with current ongoing individual deadline extensions.';
```

Comment:
Should there be a comma in front here too? Like here:
![image](https://user-images.githubusercontent.com/65202977/162943773-375eb659-53d9-414c-803e-5b45b4a4f737.png)

Reply:
I think it won't matter, the test cases will correctly pass / fail without the whole string, but no harm to put for completion